### ISSUE TITLE
Provide a history of validated recordings to user

### ISSUE THREAD

**Issue by nevik** [mozilla/voice-web#434]
I think it would be nice if a user could see a history of the recordings they validated.

Information I'd be interested in:
* "What we asked them to read"
* "What they said"
* did user approve/reject?
* timestamp

Besides being interested in this data myself, it could also be slightly extended to help with reporting problematic recordings, if the information also includes the recording's ID (or a "report" button if we want to route it through the backend).

This is relatively simple if this data is already being recorded in the backend in such a way that it can be retrieved.
If this is not the case (which I think it isn't), a server-side implementation is IMO best postponed until "proper" user accounts with logins are introduced.

Even if the backend doesn't record this data yet, we could already add client-side recording of this data, like the client already does for the two stats numbers at the top of the page.

To be clear: I'm thinking of only presenting this data to the same user who did the validations, simple for their own information (and, as mentioned above, possibly diagnostic/reporting purposes).
Data protection should still be taken into account, especially when a server-side solution comes into play. I think a client-side implementation should not be a big deal in this regard.

**Comment by mikehenrty** [comment]
I like this suggestion. Another thing we could consider doing here, is allowing users to favorite or bookmark recordings they listen to so they can go back to them later. If enough people favorite certain clips, we could even expose a clip leaderboard later.

**Comment by nevik** [comment]
I like that extension idea too, @mikehenrty!

How close do you think the project is to adding user accounts/profiles?

IOW, is it worth implementing a client-side history as I outlined above, or should we wait until we can do account-based server-side history?

**Comment by mikehenrty** [comment]
> How close do you think the project is to adding user accounts/profiles?

A month or so before we start that work. For now we have high level UX recommendations, but not designs for the new profile.

> IOW, is it worth implementing a client-side history as I outlined above, or should we wait until we can do account-based server-side history?

Two possible issues I see:
1. The client only get's the s3 public URL, which I think might expire.
1. How would this history fit in with the profile page?

**Comment by nevik** [comment]
> Two possible issues I see:
> 
> 1. The client only get's the s3 public URL, which I think might expire.
> 2. How would this history fit in with the profile page?

1. In my original proposal above, I didn't intend the user to be able to play the listed recordings again, so this would not be an issue for such a simple solution. If we want to provide this extension as we discussed above, I guess we would have to extend the backend so that we can give the client some recording ID by which it could re-request it (i.e. ask the backend for an up-to-date S3 URL).

I'd suggest an incremental approach here, where we do the simple solution (without playback of history entries) first, and extend it afterwards if we like that.

2. Well, TBH I don't know exactly what you have planned with the account/profile page overall. To reiterate: when I say integrating with/adding to the account/profile page, I mean a part of the account page that only the respective user sees privately, and not a part of the profile visible to other users.
In that context, I would suggest to simply add this history as a sub-page of the "profile" section of the website.

If you think this idea has merit, I could start by drafting up a UI mock based on the current website, to illustrate how I'm imagining it.

**Comment by mikehenrty** [comment]
In general, I like the idea but am not sure how useful this would be to me and other users. If we were to implement something like this, I would say it needs to fit with the overall direction and strategy of the profile page, which @m-branson is working on right now. In an effort to avoid overlapping/wasted effort, let's wait until we get a mock-up of what the new profile page will look like. If it doesn't satisfy the direction this bug is trying to take, let's put the feedback on that.

**Comment by nevik** [comment]
alright 😉  

@m-branson, please CC me once you've got something that you can share 😄

**Comment by mbransn** [comment]
Closing as various concepts are either now live (reporting) or part of dataset quality improvement scoping (providing validation accuracy and insights back to contributors).
